recipe/meta.yaml

```
@@ -8,17 +8,29 @@ package:
 
 source:
   fn: {{ name }}-{{ version }}.tar.gz
-  url: http://lemon.cs.elte.hu/pub/sources/lemon-{{ version  }}.tar.gz
+  url: http://lemon.cs.elte.hu/pub/sources/lemon-{{ version }}.tar.gz
   sha256: {{ sha256 }}
   patches:
+   # Note: switching to the conda compilers introduced a build error with
+   # clang 4.0.2:
+   #   lemon/path.h:231:24: error: no viable conversion from
+   #  'typename PredMapPath<Digraph, NodeMap<Arc> >::RevArcIt' to 'value_type'
+   # this patch is similar to the proposition in
+   # https://stackoverflow.com/a/33641030
+   # and uses the explicit cast operator to Arc.
+    - 0001-fix-build-error-with-clang-4.patch
    # NOTE: this disables the installation of the glpk dlls
    # from the lemon build system. An issue has been raised upstream
    # for this:
    # http://lemon.cs.elte.hu/pipermail/lemon-devel/2017-May/000392.html
-    - glpk_win.diff    # [win]
+    - 0002-glpk_win.diff
+   # The generated LEMONConfig.cmake file hard-codes static
+   # library paths, even if BUILD_SHARED_LIBS=1
+   # This patch fixes that.
+    - 0003-LEMONConfig.cmake.patch
 
 build:
-  number: 1
+  number: 2
 
```

@k-dominik, good call on the `run_exports`.
I set `max_pin='x.x.x'` here because it's unknown what compatibility is expected. If this is too strict, this can be relaxed later -- when and if a new release would come out.
```suggestion
  run_exports:
    # Windows is only built with static library.
    - {{ pin_subpackage('lemon', max_pin='x.x.x') }}  # [not win]

```